libbeat/idxmgmt/std.go

```diff
@@ -360,9 +366,56 @@ func getEventCustomIndex(evt *beat.Event) string {
 		}
 	}
 
+	// index-pattern expands the index as formatted text rather than a fixed
+	// or dated string. This was introduced to support the `Index` configuration
+	// field on Filebeat inputs, so it's essentially a workaround to communicate
+	// a specific output behavior to libbeat based on the originating input
+	// (which as implemented now is a Filebeat-specific concept).
+	// If we find ourselves adding additional backchannels like this in Meta,
+	// we should probably add a more formal (non-MapStr) configuration struct to
+	// beat.Event that inputs can use to modulate output behavior.
+	if tmp := evt.Meta["index-pattern"]; tmp != nil {
+		if pattern, ok := tmp.(*fmtstr.EventFormatString); ok {
+			idx, err := expandIndexPattern(pattern, evt.Timestamp, beatInfo)
+			if err == nil {
+				return idx
+			}
+			// Should we log a warning here somehow?
+		}
+	}
+
 	return ""
 }
 
+// Expand the given pattern string as a formatted text field with access to
+// the event's agent and timestamp.
+// This helper mimicks applyStaticFmtstr in ilm.go, creating a placeholder
+// event for the restricted set of fields we allow here. It might be worth
+// making this a shared helper function or otherwise specializing for this case.
+func expandIndexPattern(
```

Comment:
We have this pattern in a few places. When 7.0 was in the making, devs did forget to adapt one or the other place. Maybe we can provide a common formatter type/instance that can provide these default fields?